STYLEGUIDE.md

```diff
@@ -0,0 +1,367 @@
+# Design Systems Style Guide
+
+## Table of Contents
+
+- [To Contribute](#To-Contribute)
+- [Typing](#Typing)
+- [Functions](#Functions)
+- [Variable naming](#Variable-naming)
+- [Variable placement](#Variable-placement)
+- [Whitespace and line breaks](#Whitespace-and-line-breaks)
+- [JSX/React](#JSX-React)
+- [CSS-in-JS](#CSS-in-JS)
+- [File Structure](https://github.com/mongodb/leafygreen-ui/blob/main/stories/Folder-Structure.stories.mdx)
+- [References](#References)
+
+## <a id="To-Contribute">To Contribute</a>
+
+To propose updates to this style guide, please make the relevant changes in a branch and submit a PR. We review these PRs synchronously in our weekly Team Code Review and collectively gavel on new standards there.
+
+### Format Contributions
+
+### Prefer [Rule]
+
+#### Why
+
+_Description_
+
+#### Prefer
+
+_Example_
+
+#### Avoid
+
+_Example_
+
+---
+
+## <a id="Typing">Typing</a>
+
+### Prefer constants and types over enums.
+
+#### Why
+
+Ensures that TypeScript doesn't prevent people from using the values directly rather than consuming the enum.
+
+#### Prefer
+
+```typescript
+export const someConstant = {
+  optionOne = 'option 1',
+  optionTwo = 'option 2',
+} as const;
+```
+
+#### Avoid
+
+```typescript
+enum someConstant = {
+ optionOne: 'option 1',
+ optionTwo: 'option 2',
+}
+```
+
+---
+
+### When creating TS entities, place them at the top of the file, above JS
+
+---
+
+## <a id="Functions">Functions</a>
+
+### Prefer ternaries over simple `if/else` blocks
```

Comment:
But we should prefer `if/else` blocks over complex ternaries, because those are hard to read. 

#### Prefer
```ts
if (isBlah) {
 return 'a'
} else if (isFoo) {
  return 'b'
} else {
  return 'c'
}
```

over

```ts
return isBlah ? 'a' : isFoo ? 'b' : 'c'
```

Reply:
yes absolutely. I can document this as well.

Reply:
updated!